### DISCUSSION TITLE
Auto Archiving feature

### DISCUSSION THREAD

**Discussion by infinus** [bettar/miele-lxiv#111] [Ideas]
Is it possible to implement an auto archiving feature to another database? I have the main DB on flash storage and a secondary database I created on a NAS for older studies when I run out of space. 

Right now I just drag and drop between two as needed but I have to copy the files and then delete manually.

It would be nice if there was a way do something similar to database cleanup but instead of purging the files it could archive/move from the primary database to a second database based on certain metrics like remaining disk space or based on age of the files.

**Comment by infinus** [comment]
To further elaborate this would branch off the database auto cleaning logic but instead would have an option to move. 

I would propose leaving the auto cleaning option and having a different but similar section for moving this way you could use both features vs just move or delete which may be limiting to some. Although that could cause conflicts if configured incorrectly. 

 My vision would be that it would have the exact same options as delete but just replace the word delete with move and allow the user to select another secondary database.
 
I suppose this could be done at a file level but ideally using a secondary dicom node so you could query and retrieve from the archive as needed or pull comparison studies that are not in the current DB. 

This basically combines some of the existing features like auto-cleaning, routing, and on-demand but all managed within the app instead of needing to set up a secondary system.

**Comment by bettar** [comment]
How about defining your database location in a folder that is covered by iCloud syncing ?

**Reply by infinus** [reply]
My files are lager than what I could possibly keep in iCloud. I also like to keep things local. I've experimented in the past with using S3 storage mounted via external applications or things like google drive. In the end I just ended up purchasing a NAS and keeping things local as the cloud hosted files never worked well and became expensive to maintain due to cost of storage and bandwidth.
